sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py

```diff
@@ -156,12 +157,19 @@ def test_native_source(self):
 
   @attr('IT')
   def test_iobase_source(self):
+    query = StaticValueProvider(str, self.query)
     with beam.Pipeline(argv=self.args) as p:
       result = (
           p | 'read' >> beam.io._ReadFromBigQuery(
               query=self.query, use_standard_sql=True, project=self.project))
       assert_that(result, equal_to(self.TABLE_DATA))
 
+    with beam.Pipeline(argv=self.args) as p:
```

Comment:
Do we need an integration test to cover your value provider's logic? Take a look at `TestReadFromBigQuery` test case in `bigquery_test.py`. Those are unit tests for `_ReadFromBigQuery` that actually run very fast, because pipeline is not executed.

Adding new tests in `bigquery_read_it_test.py` causes huge overhead, since they are all executed by three runners (Direct, Dataflow and Flink) on every supported Python version.

Reply:
To actually evaluate the value providers, the pipeline needs to be executed right? Cause these are realistically only 'known' at runtime. The tests in `TestReadFromBigQuery` just validate the input data to the initialisation.

We can of course just validate that query is of type `ValueProvider` after initialising a pipeline, but I figured we want to check that the value provider query is actually evaluated to a string type on pipeline execution.

Am I perhaps missing something?

Reply:
What I would say is use the value provider in the first pipeline of this method, and don't add a second pipeline. I think that should be fine. WDYT @kamilwu ?

Reply:
@EDjur Thanks for explanation. That's right, if we want to check that query is successfully evaluated to a string, we have to execute the pipeline.
@pabloem I think this is good idea. Let's do it this way.